### ISSUE TITLE
Problem about training and sampling with the flow-matching model

### ISSUE THREAD

**Issue by Saoge123** [AlgoMole/MolFM#2]
Hi:
Thanks for your greate work! There are some problems when we use your code  [(supplementary materials)](https://openreview.net/attachment?id=hHUZ5V9XFu&name=supplementary_material)  to train a model and generate molecules. 
1. The performance of the flow-matching (FM) model seems to be inconsistent with the statements in your paper. Such as the following figure, the FM model (cyan) is much worse than the EDM (puple). Although the number of train steps of FM model is much less than EDM, the trend of metrics is clear.
![1705995070550](https://github.com/AlgoMole/MolFM/assets/45249977/6133178a-11b4-44bf-9f6e-fa6cf3155e47)

2. We also try to generate some molecules, but the ckpt of pretrained FM model can not be found in your code.

**Comment by Saoge123** [comment]
the training configuration in the following:
args.n_epochs = 3000
args.exp_name =  'edm_qm9'
args.n_stability_samples = 1000
args.diffusion_noise_schedule = 'polynomial_2'
args.diffusion_noise_precision = 1e-5
args.diffusion_steps = 1000
args.diffusion_loss_type = 'l2'
args.batch_size = 64
args.nf = 256
args.n_layers = 9
args.lr = 1e-4
args.normalize_factors = [1,4,10]
args.test_epochs = 20
args.ema_decay = 0.9999
args.probabilistic_model = 'flow_matching'
args.node_classifier_model_ckpt = ''

**Comment by jingjing-gong** [comment]
the checkpoints path: efm_gen/checkpoints/generative_model_ema_0.npy

**Comment by jingjing-gong** [comment]
We are still making effort to re-format the code to make it more readable. We will also release reproducible training script by then.

**Comment by Saoge123** [comment]
thanks for your quickly reply, args.pickle is needed when we run eval_sample.py.
![1705999097448](https://github.com/AlgoMole/MolFM/assets/45249977/55f64dec-a948-48b2-bdd1-06901a9c6393)

**Comment by jingjing-gong** [comment]
> thanks for your quickly reply, args.pickle is needed when we run eval_sample.py. ![1705999097448](https://private-user-images.githubusercontent.com/45249977/298866423-55f64dec-a948-48b2-bdd1-06901a9c6393.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwMDE1MzUsIm5iZiI6MTcwNjAwMTIzNSwicGF0aCI6Ii80NTI0OTk3Ny8yOTg4NjY0MjMtNTVmNjRkZWMtYTk0OC00OGIyLWJkZDEtMDY5MDFhOWM2MzkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIzVDA5MTM1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThlZjI3YjkwMWVmNGExYTVlYmUxZjgyNmZiNTUzYjhlNTcwNzU5NDdjNTVmODU0YzdmNjEyZDg0NWU1NDQwNjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3v87F5GdgHhvdLyXlxoBd83bmlqMjMWSAFLE0VRkMJM)

Sorry for the sloppiness, we have uploaded an `args.pickle` to google drive, here is the link: https://drive.google.com/file/d/1ebAcJ79AMeYq1uzcmcnVBUFIYn92--nt/view?usp=drive_link

Hope you find it useful :-)

**Comment by jingjing-gong** [comment]
Is the checkpoint working out for you?

**Comment by Saoge123** [comment]
Thanks very much for your help! The sampling code is working out, we are looking forward to your re-formated code for training.

**Comment by Frankie123421** [comment]
Hi, thanks for your nice work. I've trained the model on qm9 datasets using the pre-released code of "supplementary materials" and the hyperparameters from the "args.pickle", and I found that the performance is still worse than expected. Specifically, the validity and molecule stability converge to approximately 0.87 and 0.77 respectively. Even though the number of epochs is only around 720, it appears that further training may result in either no improvement or only marginal gains, based on the observation of the curve trend. (However, the checkpoint is at around 3000 epochs?) Could you please provide some tips on how to further enhance the performance? Thanks.

<img width="1695" alt="iShot_2024-02-04_10 01 32" src="https://github.com/AlgoMole/MolFM/assets/54012361/b3cffc9b-ad96-46bc-a7af-63b9e542b207">
